system/lib/wasmfs/wasmfs.cpp

```diff
@@ -9,25 +9,35 @@
 #include <emscripten/emscripten.h>
 #include <emscripten/html5.h>
 #include <stdlib.h>
+#include <vector>
 #include <wasi/api.h>
 
 extern "C" {
 
-int emscripten_wasmfs_printbuffer(__wasi_fd_t fd, const uint8_t* ptr, __wasi_size_t len);
+std::vector<char> buffer;
```

Comment:
```suggestion
static std::vector<char> buffer;
```
Without this, this symbol can be seen by other compilation units and with such a common name, it seems risky to collide.

Reply:
I think a comment would be good here, or a clearer name for this variable. Right now this file is small, but once it becomes large, "buffer" wouldn't be clear enough I suspect - we'll have all sorts of buffers.

Reply:
I agree, I think a clearer name change is the best choice

Reply:
Just FYI, using `static` here is more important that whatever name you choose.  `static` (or anonymous namespace) is essential IMHO.